packages/flutter/lib/src/rendering/proxy_sliver.dart

```diff
@@ -399,3 +400,39 @@ class RenderSliverAnimatedOpacity extends RenderProxySliver with RenderAnimatedO
     child = sliver;
   }
 }
+
+/// Applies a cross-axis constraint to its sliver child.
+///
+/// This render object takes an [maxExtent] parameter and uses the smaller of
+/// [maxExtent] and the parent's [SliverConstraints.crossAxisExtent] as the
+/// cross axis extent of the [SliverConstraints] passed to the sliver child.
+class RenderSliverConstrainedCrossAxis extends RenderProxySliver {
+  /// Creates a render object that constrains the cross axis extent of its sliver child.
+  ///
+  /// The [maxExtent] parameter must not be null and must be between 0 and the
+  /// parent's cross axis extent.
+  RenderSliverConstrainedCrossAxis({
+    required double maxExtent
+  }) : _maxExtent = maxExtent,
+       assert(maxExtent >= 0.0 && maxExtent < double.infinity);
+
+  /// The cross axis extent to apply to the sliver child.
+  ///
+  /// This value must be between 0 and the parent's crossAxisExtent.
+  double get maxExtent => _maxExtent;
+  double _maxExtent;
+  set maxExtent(double value) {
+    if (_maxExtent == value) {
+      return;
+    }
+    _maxExtent = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void performLayout() {
+    assert(child != null);
```

Comment:
We should still have an assertion on the value of maxExtent. One for the constructor only applies when it is created, here catches if it updated later with an invalid value.

Reply:
Done!